### ISSUE TITLE
Clarify sections on arithmetic divide and remainder

### ISSUE THREAD

**Issue by amaiorano** [gpuweb/gpuweb#3647]
In the table for binary arithmetic expressions under [7.9](https://gpuweb.github.io/gpuweb/wgsl/#arithmetic-expr), there are inconsistencies in the way divide and remainder are defined:

### Divide

* For signed integers, we say: `truncate(x) otherwise, where x is the real-valued quotient e1 ÷ e2.`
* For unsigned integers, we say: `Otherwise, the integer q such that e1 = q × e2 + r, where 0 ≤ r < e2.`

Although both are correct, they are inconsistent. I would suggest copying the one for signed integers to unsigned integers.

* For floating point: this is not defined, and should be as "x, where x is the real-valued quotient e1 ÷ e2"

### Remainder (Modulo)

* For signed integers, we say: `Otherwise, e1 - truncate(e1 ÷ e2) × e2 where the quotient is computed as a real value.`
* For unsigned integers, we say: `Otherwise, the integer r such that e1 = q × e2 + r, where q is an integer and 0 ≤ r < e2.`

For the unsigned integers case, we haven't defined what `q` is, just its range. We should say that "q is the real-valued quotient e1 ÷ e2", but my preference would be to copy the version for signed integers, defining it in terms of the truncate operation.

* For floating point, we say: ` the result is equal to: e1 - e2 * trunc(e1 / e2)`

This is fine, though we use `trunc` instead of `truncate`, which doesn't match the definitions for integrals above. We also use the forward slash `/` instead of the division symbol `÷` here. For consistency, I would suggest using the same definition as for integers: `e1 - truncate(e1 ÷ e2) × e2`.

**Comment by dneto0** [comment]
Yeah.  I used the Euclidean division definition for the unsigned case.
https://en.wikipedia.org/wiki/Euclidean_division 
That's the one the ancient Greeks used, before real numbers were invented or really understood (or believed in).  It's the more primitive definition, and used in proofs about arithmetic.

But I agree, the inconsistency is jarring, and the truncate-real-division definition is probably easier for most programmers to absorb. I can always put the euclidean definition in a note.

For the unsigned case it the condition would be   0 <= |r| < |e2| and  r = 0 or has the same sign as e2.
